common/unsafe/src/main/java/org/apache/spark/unsafe/memory/HeapMemoryAllocator.java

```diff
@@ -47,23 +48,29 @@ private boolean shouldPool(long size) {
 
   @Override
   public MemoryBlock allocate(long size) throws OutOfMemoryError {
-    if (shouldPool(size)) {
+    long alignedSize = ByteArrayMethods.roundNumberOfBytesToNearestWord(size);
```

Comment:
Maybe minor but some small allocations will be counted for pooling mechanism but they are not before, e.g. `POOLING_THRESHOLD_BYTES` - 1.

Reply:
yeah，I think it's acceptable